doc/architectural/compilation-and-development.md

```diff
@@ -17,6 +17,10 @@ To be documented.
 
 To be documented.
 
+## Running tests
+
+To be documented. (regression tests with test.pl and unit tests with Catch.)
```

Comment:
There is (partial) information in doc/architectural/cprover-architecture-overview.md, when discussing the folder `regression`. I think this should be cleaned up in this PR?

Reply:
Thanks for pointing that out - I hadn't spotted it. I've pushed a new version of this PR which updates the folder walkthrough as well, so we aren't documenting the same thing in two places. I think that documenting how to run the tests is too important to put it in the folder walkthrough, so I've kept it in the Compilation and Development page and put a link to it from the folder walkthrough.